### PR TITLE
Refactor/chat and contact subs

### PR DESCRIPTION
**Pull request by yenda**
blcoked by https://github.com/status-im/status-react/pull/6774
if you review now, focus on the second commit only

### Summary:

- get rid of message groups
- move old subs to deprecated-subs namespaces to minimize refactoring in view
and leave it for future PR
- compute active chats dynamically
- memoize expensive operations (gfycat, identicon, day-relative)

### Review notes (optional):

chat.db and contact.db namespaces are the most important parts to review

### Testing notes 

This PR requires a small sanity check to see if no error was introduces in chats

#### Platforms
- Android
- iOS
- macOS
- Linux
- Windows

#### Areas that maybe impacted (optional)
**Functional**
- 1-1 chats
- public chats
- group chats
- wallet / transactions

**Non-functional**
- CPU performance / speed of the app

status: ready

### CODE REVIEW

File: src/status_im/chat/deprecated_subs.cljs [hunk 1]
```diff
@@ -0,0 +1,84 @@
+(ns status-im.chat.deprecated-subs
```
Comment (rcullito) [inline src/status_im/chat/deprecated_subs.cljs:1]: Trying to follow along here. Is the current situation that the new chat.db and contact.db modifications will allow for new features (such as blocking) that were not possible before? Yet migrating all of the app over to use those would be such a massive undertaking that it is outside the scope of any one PR and therefore this namespace, *deprecated-subs* is needed?

Comment (yenda) [reply src/status_im/chat/deprecated_subs.cljs:1]: yes the new subs are computing the active chats and all metadata associated dynamically, including the list of ordered messages, datemarks and metadata. but to avoid 500 extra lines of code I move the subs old subs to deprecated-subs namespace and will update the views in future updates.

Comment (rcullito) [reply src/status_im/chat/deprecated_subs.cljs:1]: 👍 thanks for clarifying.
